workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -129,5 +130,214 @@ private static bool TryGetApkPath(string rootPath, out string apkPath)
+
+                var toolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+                if (string.IsNullOrEmpty(toolsConfig.DevelopmentTeamId))
+                {
+                    Debug.LogError("Development Team Id was not specified. Unable to build the XCode project.");
+                    return;
+                }
+
+                if (RedirectedProcess.Command("xcodebuild")
+                    .WithArgs("build-for-testing",
+                        "-project", Path.Combine(XCodeProjectPath, XCodeProjectFile),
+                        "-derivedDataPath", DerivedDataPath,
+                        "-scheme", "Unity-iPhone",
+                        $"DEVELOPMENT_TEAM={toolsConfig.DevelopmentTeamId}",
+                        "-allowProvisioningUpdates")
+                    .Run() != 0)
+                {
+                    Debug.LogError(
+                        $"Failed to build your XCode project. Make sure you have xcodebuild installed and check the logs.");
+                    return;
+                }
+
+                var deviceUID = string.Empty;
+                var xcTestRunPath = string.Empty;
+
+                // Check if we have a physical device connected
+                var availableDevices = new List<string>();
+                var deviceRegex = new Regex("\\[([a-z]|[0-9])+\\]");
+                RedirectedProcess.Command("instruments")
+                    .WithArgs("-s", "devices")
+                    .AddOutputProcessing(message =>
+                    {
+                        if (deviceRegex.IsMatch(message))
+                        {
+                            availableDevices.Add(deviceRegex.Match(message).Value);
+                        }
+                    })
+                    .Run();
+
+                if (!TryGetXCTestRunPath(availableDevices.Count == 0, out xcTestRunPath))
+                {
+                    Debug.LogError("Unable to fund a xctestrun file for the correct architecture. Did you built it for the correct platform?");
+                    return;
+                }
+
+                if (!TryPrepareArguments(shouldConnectLocally, out var arguments))
+                {
+                    Debug.LogError("Failed to generate arguments");
+                    return;
+                }
+
+                // add SpatialOS arguments to iOS environment variables
+                var doc = new XmlDocument();
+                doc.Load(xcTestRunPath);
+                XmlNode spatialNode = null;
+                var rootNode = doc.DocumentElement.ChildNodes[0].ChildNodes[1];
+                for (var i = 0; i < rootNode.ChildNodes.Count; i++)
+                {
+                    var node = rootNode.ChildNodes[i];
+                    if (node.InnerText != "EnvironmentVariables")
+                    {
+                        continue;
+                    }
+
+                    var envValueNode = rootNode.ChildNodes[i + 1];
```

Comment:
Can this go out of bounds?

Reply:
if it's a misconfigured xctestrunfile, otherwise no. The value of a key is always in the next node. we can surround it with a try-catch and throw an error that something is wrong with the xctestrun file